ilastik/applets/pixelClassification/suggestFeaturesDialog.py

```diff
@@ -767,157 +777,186 @@ def _auto_select_num_features(self, feature_order):
-            selected_ids = self._convert_featureMatrix_to_featureIDs(user_defined_matrix)
-            current_features_result = FeatureSelectionResult(
-                user_defined_matrix,
-                selected_ids,
-                segmentation_current_features,
-                {"num_of_feat": "user", "c": "None"},
-                "user features",
-                oob_user,
-                time_user,
+                self.opWrapperFeatureSelection.ComplexityPenalty.setValue(self._selection_params["c"])
+                selected_feature_ids = self.opWrapperFeatureSelection.SelectedFeatureIDs.value
+
+            # create a new layer for display in the volumina viewer
+            # make sure to save the feature matrix used to obtain it
+            # maybe also write down feature computation time and oob error
+            new_matrix = self._convert_featureIDs_to_featureMatrix(selected_feature_ids)
+            new_segmentation, new_oob, new_time = self.retrieve_segmentation(new_matrix)
+            new_feature_selection_result = SuggestFeaturesResult(
+                new_matrix,
+                selected_feature_ids,
+                new_segmentation,
+                self._selection_params,
+                self._selection_method,
+                oob_err=new_oob,
+                feature_calc_time=new_time,
             )
-            self._add_feature_set_to_results(current_features_result)
-            self._initialized_current_features_segmentation_layer = True
+            self._add_feature_set_to_results(new_feature_selection_result)
+
+            if not self._initialized_current_features_segmentation_layer:
+                # FIXME: this should probably be moved somewhere else
+                self.opFeatureSelection.setupOutputs()  # deletes cache for realistic feature computation time
+                segmentation_current_features, oob_user, time_user = self.retrieve_segmentation(user_defined_matrix)
+                selected_ids = self._convert_featureMatrix_to_featureIDs(user_defined_matrix)
+                current_features_result = SuggestFeaturesResult(
+                    user_defined_matrix,
+                    selected_ids,
+                    segmentation_current_features,
+                    {"num_of_feat": "user", "c": "None"},
+                    "user features",
+                    oob_user,
+                    time_user,
+                )
+                self._add_feature_set_to_results(current_features_result)
+                self._initialized_current_features_segmentation_layer = True
 
-        # revert changes to matrix
-        self.opFeatureSelection.SelectionMatrix.setValue(user_defined_matrix)
-        self.opFeatureSelection.SelectionMatrix.setDirty()  # this does not do anything!?!?
-        self.opFeatureSelection.setupOutputs()
-        QtWidgets.QApplication.instance().restoreOverrideCursor()
+        finally:
```

Comment:
I wonder if a context manager that reverts slot values on `__exit__` would be more readable and robust.

Reply:
agreed, I'll give it a go and see how it looks. Thx! :)